### PR TITLE
Remove exec from weaver command

### LINKED ISSUE #2836: Zombie apocalypse with kubernetes and weave-kube
Kubernetes:

    $ kubectl version --short
    Client Version: v1.5.4
    Server Version: v1.5.4

Weave:

    $ kubectl get ds -n kube-system weave-net -o jsonpath='{.spec.template.spec.containers[*].image}'
    weaveworks/weave-kube:1.9.3 weaveworks/weave-npc:1.9.3

Zombies:

    $ knife ssh "role:kubernetes_node" "sudo ps aux|grep [l]aunch.sh"|sort
    kube01 root      2230  0.0  0.0      0     0 ?        Z    Mar07   0:00 [launch.sh] <defunct>
    kube02 root      7411  0.0  0.0      0     0 ?        Z    Mar07   0:00 [launch.sh] <defunct>
    kube03 root      5007  0.0  0.0      0     0 ?        Z    Mar07   0:00 [launch.sh] <defunct>
    kube04 root      6274  0.0  0.0      0     0 ?        Z    19:19   0:00 [launch.sh] <defunct>
	...

`uname -a`:

    $ knife ssh "role:kubernetes_node" "uname -a"|sort
    kube01 Linux kube01 4.4.0-51-generic #72-Ubuntu SMP Thu Nov 24 18:29:54 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
    kube02 Linux kube02 4.4.0-51-generic #72-Ubuntu SMP Thu Nov 24 18:29:54 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
    kube03 Linux kube03 4.4.0-51-generic #72-Ubuntu SMP Thu Nov 24 18:29:54 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
    kube04 Linux kube04 4.4.0-64-generic #85-Ubuntu SMP Mon Feb 20 11:50:30 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

Maybe it is connected with following issue in kubernetes: https://github.com/kubernetes/kubernetes/issues/39334
We experience this cluster-wide.

### PR DESCRIPTION
**Pull request by marccarre**
Fixes #2836. 
Maybe more a workaround than a real fix? (see #2844).
However it seems to do the job -- see below manual testing.

### CODE REVIEW

File: test/840_weave_kube_3_test.sh [hunk 1]
```diff
@@ -52,6 +52,12 @@ assert_raises "run_on $HOST1 $PING $HOST2EXPIP"
 assert_raises "run_on $HOST2 $PING $HOST1EXPIP"
 assert_raises "run_on $HOST3 $PING $HOST2EXPIP"
 
+# Check launch.sh does not stick around as a defunct process after having started weaver:
+count_defunct_launch_processes="ps aux | grep 'launch.sh' | grep defunct | wc -l"
```
Comment (bboreham) [inline test/840_weave_kube_3_test.sh]: can we just check for any defunct processes, i.e. not require them to be named `launch.sh`?

also `grep -c` would elide `wc -l`

**Comment by bboreham** [comment]
Can you show the output of `ps auxf` here, like you did in #2844?

**Comment by marccarre** [comment]
As requested, here is the output of `ps auxf`:

- during initialisation:

```
$ ps auxf
[...]
root      1380  0.4  1.9 879660 74660 ?        Ssl  Mar13   6:02 /usr/bin/docker daemon -H fd:// -H unix:///
root      1664  0.0  0.4 502116 18528 ?        Ssl  Mar13   0:06  \_ docker-containerd -l /var/run/docker/li
root     15482  0.0  0.0 200496  3364 ?        Sl   15:04   0:00      \_ docker-containerd-shim 3d1f5eb6e090
root     15496  0.0  0.0   1524   996 ?        Ss   15:04   0:00      |   \_ /bin/sh /home/weave/launch.sh
root     15780  0.0  0.0   1524    64 ?        S    15:04   0:00      |       \_ /bin/sh /home/weave/launch.
root     15872  0.0  0.0   1772  1272 ?        S    15:04   0:00      |       |   \_ /bin/sh /home/weave/wea
root     15897  0.0  0.0   1772   324 ?        S    15:04   0:00      |       |       \_ /bin/sh /home/weave
root     15898  0.0  0.0  14656  2948 ?        S    15:04   0:00      |       |           \_ curl -o /tmp/we
root     15781 10.0  1.5 484556 60296 ?        Sl   15:04   0:00      |       \_ /home/weave/weaver --port=6
[...]
```

- after initialisation:

```
$ ps auxf
[...]
root      1380  0.4  1.9 879660 74660 ?        Ssl  Mar13   6:02 /usr/bin/docker daemon -H fd:// -H unix:///
root      1664  0.0  0.4 502116 18592 ?        Ssl  Mar13   0:06  \_ docker-containerd -l /var/run/docker/li
root     15482  0.0  0.0 200496  3364 ?        Sl   15:04   0:00      \_ docker-containerd-shim 3d1f5eb6e090
root     15496  0.0  0.0   1524   996 ?        Ss   15:04   0:00      |   \_ /bin/sh /home/weave/launch.sh
root     15781  2.3  1.6 503064 64320 ?        Sl   15:04   0:00      |       \_ /home/weave/weaver --port=6[...]
[...]
```

I've also added the above to the commit message's details, for posterity.

**Comment by marccarre** [comment]
I've also just tried to `docker kill` the container and I cannot see any `defunct` process like mentioned in #2684.

**Comment by marccarre** [comment]
Neither does a regular `kill` against `/bin/sh /home/weave/launch.sh`.

**Comment by marccarre** [comment]
Note that I could **not** pinpoint what is responsible for such behaviour, i.e. "no orphan `weaver` after `SIGKILL` on `launch.sh`", which I did not expect though it seems to "just work". 
A couple observations:
- no process should be able to handle `SIGKILL`, to then propagate it;
- `launch.sh` and `weaver` appear to be in the same process group, so maybe the signal is sent to the group instead:
```
$ ps axo pid,pgid,comm | grep -B1 weaver
 8675  8675 launch.sh
 8737  8675 weaver
```
- as mentioned on #2684, it could be that `PR_SET_PDEATHSIG` is set on the shell.

**Pull request merged**